### PR TITLE
Sidekiq-ized badge jobs

### PR DESCRIPTION
**Pull request by nickinack**
Added jobs as instructed.

### CODE REVIEW

File: app/models/submission.rb [hunk 1]
```diff
@@ -56,9 +56,7 @@ class Submission < ApplicationRecord
         .perform_later(challenge_round_id: challenge_round_id)
     end
     Prometheus::SubmissionCounterService.new(submission_id: id).call
-    for badge in AicrowdBadge.where(badges_event_id: BadgesEvent.where(name: "onsubmission").pluck(:id))
-      eval(badge.code)
-    end
+    ParticipantBadgeJob.perform_later(name: "onsubmission")
```
Comment (skbly7) [inline app/models/submission.rb:59]: Doesn't you need to pass ID or something, otherwise job can't know which submission/profile has been updated?

Comment (nickinack) [reply app/models/submission.rb]: Resolved

File: app/models/submission.rb [hunk 2]
```diff
@@ -56,9 +56,8 @@ class Submission < ApplicationRecord
         .perform_later(challenge_round_id: challenge_round_id)
     end
     Prometheus::SubmissionCounterService.new(submission_id: id).call
-    for badge in AicrowdBadge.where(badges_event_id: BadgesEvent.where(name: "onsubmission").pluck(:id))
-      eval(badge.code)
-    end
+    participant_id = Participant.find_by(id: Submission.find_by(id: id).participant_id)
+    ParticipantBadgeJob.perform_later(name: "onsubmission", participant_id: participant_id, grading_status_cd: grading_status_cd)
```
Comment (skbly7) [inline app/models/submission.rb:60]: You should be able to directly use `participant_id: participant.id` and skip the above line completely i.e.

```
participant_id = Participant.find_by(id: Submission.find_by(id: id).participant_id)
```

Comment (nickinack) [reply app/models/submission.rb:60]: Sure then!

File: app/models/submission.rb [hunk 3]
```diff
@@ -56,9 +56,7 @@ class Submission < ApplicationRecord
         .perform_later(challenge_round_id: challenge_round_id)
     end
     Prometheus::SubmissionCounterService.new(submission_id: id).call
-    for badge in AicrowdBadge.where(badges_event_id: BadgesEvent.where(name: "onsubmission").pluck(:id))
-      eval(badge.code)
-    end
+    ParticipantBadgeJob.perform_later(name: "onsubmission", participant_id: participant.id, grading_status_cd: grading_status_cd)
```
Comment (skbly7) [inline app/models/submission.rb:59]: `submission_id`?

**Pull request closed without merging**